app/controllers/issue_templates_controller.rb

```diff
@@ -1,206 +1,219 @@
-
-    @globalIssueTemplates = GlobalIssueTemplate.joins(:projects).where(["tracker_id = ? AND projects.id = ?",
-                                                                        @tracker.id, @project.id]).order('position')
-
-
-    if @globalIssueTemplates.any?
-      @globalIssueTemplates.each do |x|
-        group.push([x.title, x.id, {:class => "global"}])
-        # Using global template as default template is now disabled.
-        # if x.is_default == true
-        #   if project_default_template.blank?
-        #     @default_template = x
-        #   end
-        # end
-      end
-    end
-
-    @grouped_options.push([@tracker.name, group]) if group.any?
-    render :action => "_template_pulldown", :layout => false
-  end
-
-  # preview
-  def preview
-    @text = (params[:issue_template] ? params[:issue_template][:description] : nil)
-    @issue_template = IssueTemplate.find(params[:id]) if params[:id]
-    render :partial => 'common/preview'
-  end
-
-  # Reorder templates
-  def move
-    move_order(params[:to])
-  end
-    
-  private
-  def find_user
-    @user = User.current
-  end
-  
-  def find_tracker
-    @tracker = Tracker.find(params[:issue_tracker_id])
-  end
-
-  def find_object
-    @issue_template = IssueTemplate.find(params[:id])
-    @project = @issue_template.project
-  rescue ActiveRecord::RecordNotFound
-    render_404
-  end
-  
-  def find_project
-    begin
-      @project = Project.find(params[:project_id])
-    rescue ActiveRecord::RecordNotFound
-      render_404
-    end
-  end
-
-  def move_order(method)
-    IssueTemplate.find(params[:id]).send "move_#{method}"
-    respond_to do |format|
-      format.html { redirect_to :action => 'index' }
-      format.xml  { head :ok }
-    end
-  end
-end
-
+# noinspection ALL
+class IssueTemplatesController < ApplicationController
+  unloadable
+  layout 'base'
+  include IssueTemplatesHelper
+  helper :issues
+  include IssuesHelper
+  menu_item :issues
+  before_filter :find_object, only: [:show, :edit, :destroy]
+  before_filter :find_user, :find_project, :authorize,
+                except: [:preview, :move_order_higher, :move_order_lower, :move_order_to_top, :move_order_to_bottom, :move]
+  before_filter :find_tracker, only: [:set_pulldown]
+
+  def index
+    tracker_ids = IssueTemplate.search_by_project(@project.id).pluck(:tracker_id).uniq
+
+    @template_map = {}
+    tracker_ids.each do |tracker_id|
+      templates = IssueTemplate.search_by_project(@project.id).search_by_tracker(tracker_id).order_by_position
+      @template_map[Tracker.find(tracker_id)] = templates if templates.any?
+    end
+
+    @setting = IssueTemplateSetting.find_or_create(@project.id)
+    inherit_template = @setting.enabled_inherit_templates?
+    @inherit_templates = []
+
+    project_ids = inherit_template ? @project.ancestors.collect(&:id) : [@project.id]
+    if inherit_template
+      # keep ordering
+      used_tracker_ids = @project.trackers.pluck(:tracker_id)
+=begin
```

Comment:
__[rubocop]__
* Do not use block comments. [:ref](https://github.com/bbatsov/ruby-style-guide#no-block-comments)